aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -289,6 +291,27 @@ Tensor _s_where(const Tensor& condition, const Tensor& self, const Tensor& other
   return ret;
 }
 
+void zero_numel_tensor_resize(Tensor& result, Tensor& result_indices, const Tensor& self, int64_t dim,
+                              bool keepdim) {
+  TORCH_CHECK(self.size(dim) != 0, "Expected reduction dim ", dim, " to be non-zero");
+
+  if (keepdim) {
+    auto sizes = ensure_nonempty_vec(self.sizes().vec());
+    sizes[dim] = 1;
+    result.resize_(sizes);
```

Comment:
What advantage would that have over the check that is performed by `resize_output`? Most of the sample code I read in other comparison functions used `resize_`. If that's the new recommended way of resizing tensors I can change it.

Reply:
There's a fairly new effort to use `resize_output` for result tensors of out-variant functions, for better consistency and error checking: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Resize.h#L11

Only 2 out of 3 of your `zero_numel_tensor_resize` call sites give a result tensor, right (`max_out_impl` and `_aminmax_out_impl`)? I guess it would be more correct to only use `resize_output` in those cases, and `resize_` otherwise. But maybe it's just simpler to leave it as you have it, for now.

@ezyang, do you have a strong opinion either way?